modules/javafx.graphics/src/main/java/javafx/scene/text/CaretInfo.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javafx.scene.text;
+
+import com.sun.javafx.text.PrismCaretInfo;
+
+/**
+ * Provides the information associated with the caret.
+ *
+ * @since 24
+ */
+public sealed abstract class CaretInfo permits PrismCaretInfo {
```

Comment:
Given the API you define, is there a reason not to make this an interface?

Reply:
Originally it was an interface.  Then @prrace said "Why an interface ? It is easier to evolve a class when you want to add more info."

Since this thing is sealed, it probably does not matter, and access to class methods is _two_ nanoseconds faster (not that's important).

The only good thing about interface is the ease of mocking in testing, but we don't do that.